oneview/data_source_logical_enclosure.go

```diff
@@ -0,0 +1,244 @@
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+	d.SetId("name")
+	d.Set("ambient_temperature_mode", logicalEnclosure.AmbientTemperatureMode)
+	d.Set("category", logicalEnclosure.Category)
+	d.Set("created", logicalEnclosure.Created)
+	d.Set("delete_failed", logicalEnclosure.DeleteFailed)
+	deploymentManagerSettings := make([]map[string]interface{}, 0, 1)
+	deploymentManagerSettings = append(deploymentManagerSettings, map[string]interface{}{
+		"deployment_mode":         logicalEnclosure.DeploymentManagerSettings.OsDeploymentSettings.DeploymentModeSettings.DeploymentMode,
+		"deployment_network_uri":  logicalEnclosure.DeploymentManagerSettings.OsDeploymentSettings.DeploymentModeSettings.DeploymentNetworkUri,
+		"manage_os_deployment":    logicalEnclosure.DeploymentManagerSettings.OsDeploymentSettings.ManageOSDeployment,
+		"deployement_cluster_uri": logicalEnclosure.DeploymentManagerSettings.DeploymentClusterUri,
+	})
+
+	d.Set("deployment_manager_settings", deploymentManagerSettings)
+	d.Set("description", logicalEnclosure.Description)
+	d.Set("enclosure_group_uri", logicalEnclosure.EnclosureGroupUri.String())
+	d.Set("enclosure_uris", logicalEnclosure.EnclosureUris)
+	logicalEnclosureFirmware := make([]map[string]interface{}, 0, 1)
+	logicalEnclosureFirmware = append(logicalEnclosureFirmware, map[string]interface{}{
+		"firmware_baseline_uri":                            logicalEnclosure.Firmware.FirmwareBaselineUri,
+		"firmware_update_on":                               logicalEnclosure.Firmware.FirmwareUpdateOn,
+		"force_install_firmware":                           logicalEnclosure.Firmware.ForceInstallFirmware,
+		"logical_interconnect_update_mode":                 logicalEnclosure.Firmware.LogicalInterconnectUpdateMode,
+		"update_firmware_on_unmanaged_interconnect":        logicalEnclosure.Firmware.UpdateFirmwareOnUnmanagedInterconnect,
+		"validate_if_li_firmware_update_is_non_disruptive": logicalEnclosure.Firmware.ValidateIfLIFirmwareUpdateIsNonDisruptive,
+	})
+	d.Set("firmware", logicalEnclosureFirmware)
+	d.Set("ip_addressing_mode", logicalEnclosure.IpAddressingMode)
+	logicalEnclosureIpv4Ranges := make([]map[string]interface{}, 0, len(logicalEnclosure.Ipv4Ranges))
+	for _, logicalEnclosureIpv4Range := range logicalEnclosure.Ipv4Ranges {
+		dnsServerMap := make([]interface{}, len(logicalEnclosureIpv4Range.DnsServers))
+		for i, dnsServer := range logicalEnclosureIpv4Range.DnsServers {
+			dnsServerMap[i] = dnsServer
+		}
+		logicalEnclosureIpv4Ranges = append(logicalEnclosureIpv4Ranges, map[string]interface{}{
+			"dns_servers":  schema.NewSet(schema.HashString, dnsServerMap),
+			"domain":       logicalEnclosureIpv4Range.Domain,
+			"gateway":      logicalEnclosureIpv4Range.Gateway,
+			"ip_range_uri": logicalEnclosureIpv4Range.IpRangeUri,
+			"name":         logicalEnclosureIpv4Range.Name,
+			"subnet_mask":  logicalEnclosureIpv4Range.SubnetMask,
+		})
+	}
+	ipv4RangeCount := d.Get("ipv4_range.#").(int)
```

Comment:
The logic around checking what's in the config file and ordering the api results based on that needs to be removed. This doesn't matter for data sources and needs to be removed as a whole from the provider